examples/pytorch_ignite_track.py

```diff
@@ -0,0 +1,180 @@
+
+# Download and load the training data
+trainset = datasets.FashionMNIST('./data', download=True, train=True, transform=transform)
+train_loader = DataLoader(trainset, batch_size=64, shuffle=True)
+
+# Download and load the test data
+validationset = datasets.FashionMNIST('./data', download=True, train=False, transform=transform)
+val_loader = DataLoader(validationset, batch_size=64, shuffle=True)
+
+
+class CNN(nn.Module):
+
+    def __init__(self):
+        super(CNN, self).__init__()
+
+        self.convlayer1 = nn.Sequential(
+            nn.Conv2d(1, 32, 3, padding=1),
+            nn.BatchNorm2d(32),
+            nn.ReLU(),
+            nn.MaxPool2d(kernel_size=2, stride=2)
+        )
+
+        self.convlayer2 = nn.Sequential(
+            nn.Conv2d(32, 64, 3),
+            nn.BatchNorm2d(64),
+            nn.ReLU(),
+            nn.MaxPool2d(2)
+        )
+
+        self.fc1 = nn.Linear(64 * 6 * 6, 600)
+        self.drop = nn.Dropout2d(0.25)
+        self.fc2 = nn.Linear(600, 120)
+        self.fc3 = nn.Linear(120, 10)
+
+    def forward(self, x):
+        x = self.convlayer1(x)
+        x = self.convlayer2(x)
+        x = x.view(-1, 64 * 6 * 6)
+        x = self.fc1(x)
+        x = self.drop(x)
+        x = self.fc2(x)
+        x = self.fc3(x)
+
+        return F.log_softmax(x, dim=1)
+
+# creating model,and defining optimizer and loss
+model = CNN()
+# moving model to gpu if available
+device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+model.to(device)
+optimizer = optim.Adam(model.parameters(), lr=0.001)
+criterion = nn.NLLLoss()
+
+
+# defining the number of epochs
+epochs = 12
+# creating trainer,evaluator
+trainer = create_supervised_trainer(model, optimizer, criterion, device=device)
+metrics = {
+    'accuracy': Accuracy(),
+    'nll': Loss(criterion),
+    'cm': ConfusionMatrix(num_classes=10)
+}
+train_evaluator = create_supervised_evaluator(model, metrics=metrics, device=device)
+val_evaluator = create_supervised_evaluator(model, metrics=metrics, device=device)
+training_history = {'accuracy':[],'loss':[]}
+validation_history = {'accuracy':[],'loss':[]}
+last_epoch = []
+
+RunningAverage(output_transform=lambda x: x).attach(trainer, 'loss')
+
+
+def score_function(engine):
+    val_loss = engine.state.metrics['nll']
+    return -val_loss
+
+
+handler = EarlyStopping(patience=10, score_function=score_function, trainer=trainer)
+val_evaluator.add_event_handler(Events.COMPLETED, handler)
+
+
+@trainer.on(Events.EPOCH_COMPLETED)
+def log_training_results(trainer):
+    train_evaluator.run(train_loader)
+    metrics = train_evaluator.state.metrics
+    accuracy = metrics['accuracy'] * 100
+    loss = metrics['nll']
+    last_epoch.append(0)
+    training_history['accuracy'].append(accuracy)
+    training_history['loss'].append(loss)
+    print("Training Results - Epoch: {}  Avg accuracy: {:.2f} Avg loss: {:.2f}"
+          .format(trainer.state.epoch, accuracy, loss))
```

Comment:
```suggestion
@trainer.on(Events.EPOCH_COMPLETED)
def validation(trainer):
    train_evaluator.run(train_loader)
    val_evaluator.run(val_loader)    
```

Let's simplify the code to the essential things. IMO, no need to print to stdout if we can see the curves.